layers/core_checks/cc_render_pass.cpp

```diff
@@ -5104,3 +5113,226 @@ bool CoreChecks::ValidateInheritanceInfoFramebuffer(VkCommandBuffer primaryBuffe
+    }
+
+    return skip;
+}
+
+bool CoreChecks::PreCallValidateCmdSetRenderingAttachmentLocationsKHR(VkCommandBuffer commandBuffer,
+                                                                      const VkRenderingAttachmentLocationInfoKHR *pLocationInfo,
+                                                                      const ErrorObject &error_obj) const {
+    const auto &cb_state = *GetRead<vvl::CommandBuffer>(commandBuffer);
+    bool skip = false;
+
+    if (!enabled_features.dynamicRenderingLocalRead) {
+        skip |= LogError("VUID-vkCmdSetRenderingAttachmentLocationsKHR-dynamicRenderingLocalRead-09509", commandBuffer,
+                         error_obj.location, "dynamicRenderingLocalRead was not enabled.");
+    }
+
+    skip |= ValidateCmd(cb_state, error_obj.location);
+
+    const auto *rp_state_ptr = cb_state.activeRenderPass.get();
+    if (!rp_state_ptr) {
+        return skip;
+    }
+
+    const auto &rp_state = *rp_state_ptr;
+
+    if (!rp_state.UsesDynamicRendering()) {
+        const LogObjectList objlist(commandBuffer, rp_state.VkHandle());
+        skip |= LogError("VUID-vkCmdSetRenderingAttachmentLocationsKHR-commandBuffer-09511", objlist, error_obj.location,
+                         "Please use vkCmdBeginRendering.");
+    }
+
+    if (pLocationInfo && pLocationInfo->colorAttachmentCount != rp_state.dynamic_rendering_begin_rendering_info.colorAttachmentCount) {
+        const LogObjectList objlist(commandBuffer, rp_state.VkHandle());
+        skip |= LogError("VUID-vkCmdSetRenderingAttachmentLocationsKHR-pLocationInfo-09510", objlist, error_obj.location,
+                         "colorAttachmentCount (%" PRIu32 ") is not equal to count specified in VkRenderingInfo (%" PRIu32 ").",
+                         pLocationInfo->colorAttachmentCount, rp_state.createInfo.attachmentCount);
+    }
+
+    skip |= ValidateRenderingAttachmentLocationsKHR(pLocationInfo, commandBuffer, error_obj.location);
+
+    return skip;
+}
+
+bool CoreChecks::ValidateRenderingInputAttachmentIndicesKHR(const VkRenderingInputAttachmentIndexInfoKHR* pLocationInfo,
+                                                            const LogObjectList objlist, const Location &loc_info) const {
+    bool skip = false;
+
+    if (!pLocationInfo) {
+        return skip;
+    }
+
+    if (!enabled_features.dynamicRenderingLocalRead) {
+        if (pLocationInfo->pColorAttachmentInputIndices) {
+            for (uint32_t i = 0; i < pLocationInfo->colorAttachmentCount; ++i) {
+                if (pLocationInfo->pColorAttachmentInputIndices[i] != VK_ATTACHMENT_UNUSED) {
+                    const Location loc =
+                        loc_info.dot(Struct::VkRenderingInputAttachmentIndexInfoKHR, Field::pColorAttachmentInputIndices, i);
+                    skip |= LogError("VUID-VkRenderingInputAttachmentIndexInfoKHR-dynamicRenderingLocalRead-09519", objlist, loc,
```

Comment:
```suggestion
                    skip |= LogError("VUID-VkRenderingInputAttachmentIndexInfoKHR-dynamicRenderingLocalRead-09519", objlist, loc_info.dot(Struct::VkRenderingInputAttachmentIndexInfoKHR, Field::pColorAttachmentInputIndices, i),
```

nit: no need to create an extra variable here (only because we don't really do that anywhere else and rather be consistent)